kernel/arch/x86_64/entry_asm.nasm

```diff
@@ -0,0 +1,18 @@
+        BITS 64
+
+        extern _entry_c
+        extern stack_top
+
+global _entry
+_entry:
+        ;; backup rax to rcx
+        mov rcx, rax
+        ;; set up the stack via rax
```

Comment:
We do it this way rather than via lea in order to avoid truncation of the `stack_top` symbol to fit `R_X86_64_32S`